### PR TITLE
Adds mob attack signals [READY]

### PR DESCRIPTION
**Pull request by yoyobatty**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: YoYoBatty
code: Added signals for various mob attacking actions.
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

### CODE REVIEW

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -116,6 +116,7 @@
 		return
 
 	if(in_throw_mode)
+		SEND_SIGNAL(src, COMSIG_MOB_THROW, A, params)
```
Comment (ninjanomnom) [inline code/_onclick/click.dm]: Put this signal in the throw_item proc

File: code/_onclick/click.dm [hunk 2]
```diff
@@ -131,10 +132,12 @@
 	if(A in DirectAccess())
 		if(W)
 			W.melee_attack_chain(src, A, params)
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK, A, params)
 		else
 			if(ismob(A))
 				changeNext_move(CLICK_CD_MELEE)
 			UnarmedAttack(A)
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK, A, params)
```
Comment (ninjanomnom) [inline code/_onclick/click.dm]: Put this signal in the UnarmedAttack proc and have a separate signal for unarmed

File: code/_onclick/click.dm [hunk 3]
```diff
@@ -145,15 +148,19 @@
 	if(CanReach(A,W))
 		if(W)
 			W.melee_attack_chain(src, A, params)
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK, A, params)
 		else
 			if(ismob(A))
 				changeNext_move(CLICK_CD_MELEE)
 			UnarmedAttack(A,1)
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK, A, params)
```
Comment (ninjanomnom) [inline code/_onclick/click.dm]: Same as above

File: code/_onclick/click.dm [hunk 4]
```diff
@@ -145,15 +148,19 @@
...
 				changeNext_move(CLICK_CD_MELEE)
 			UnarmedAttack(A,1)
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK, A, params)
 	else
 		if(W)
 			W.afterattack(A,src,0,params)
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK, A, params)
 		else
 			RangedAttack(A,params)
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK_RANGED, A, params)
```
Comment (ninjanomnom) [inline code/_onclick/click.dm]: Put this signal in the RangedAttack proc

**Review by ninjanomnom — CHANGES_REQUESTED** [review]
You may need to do some minor refactors if any of these don't call parent and you put the signal in the proc definition

File: code/_onclick/click.dm [hunk 5]
```diff
@@ -145,15 +148,19 @@
...
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK, A, params)
 		else
 			if(ismob(A))
 				changeNext_move(CLICK_CD_MELEE)
 			UnarmedAttack(A,1)
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK, A, params)
 	else
 		if(W)
 			W.afterattack(A,src,0,params)
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK, A, params)
```
Comment (ninjanomnom) [inline code/_onclick/click.dm]: Have a seperate signal for this as well

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/_onclick/click.dm (hunk 1 above)
Comment (yoyobatty) [reply code/_onclick/click.dm]: it already has a signal that's sent from the item being thrown

File: code/_onclick/click.dm (hunk 2 above)
Comment (yoyobatty) [reply code/_onclick/click.dm]: will do

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/_onclick/click.dm [hunk 6]
```diff
@@ -145,13 +146,15 @@
...
 			W.melee_attack_chain(src, A, params)
+			SEND_SIGNAL(src, COMSIG_MOB_ITEM_ATTACK, A, params)
 		else
 			if(ismob(A))
 				changeNext_move(CLICK_CD_MELEE)
 			UnarmedAttack(A,1)
 	else
 		if(W)
 			W.afterattack(A,src,0,params)
+			SEND_SIGNAL(src, COMSIG_MOB_ITEM_ATTACK, A, params)
```
[the remark this answers is not in the document]
Comment (yoyobatty) [reply code/_onclick/click.dm]: COMSIG_MOB_ITEM_ATTACK is different from COMSIG_MOB_ATTACK?

[the remark this answers is not in the document]
Comment (ninjanomnom) [reply code/_onclick/click.dm]: You're using the same signal here for both afterattack and melee_attack_chain

[the remark this answers is not in the document]
Comment (yoyobatty) [reply code/_onclick/click.dm]: kk

[the remark this answers is not in the document]
Comment (yoyobatty) [reply code/_onclick/click.dm]: Done !

**Comment by ninjanomnom** [comment]
Some of the overrides for the procs the signal are now in don't call parent I believe, you'll need to check up on that.

**Comment by yoyobatty** [comment]
So should I have them all call the parent?

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**